.circleci/config.yml

```diff
@@ -127,57 +132,140 @@ jobs:
           name: bundle install gems
           command: |
             $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
-            bundle install
+            bundle install --path vendor/bundle
 
       - save_cache:
           name: save gem cache
           paths:
             - ./vendor/bundle
           key: v1-bundle-<< parameters.ruby_version >>-{{ .Branch }}-{{ checksum "tiny_tds.gemspec" }}
 
+      - restore_cache:
+          name: restore cross-compiled code
+          keys:
+            - compiled-<< pipeline.git.revision >>
+
       - run:
-          name: build openssl
-          no_output_timeout: 30m
+          name: restore cross-compiled code
           command: |
             $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
-            bundle exec rake ports:openssl
+
+            $rubyArchitecture = (ruby -e 'puts RUBY_PLATFORM').Trim()            
+            $source = "C:\home\circleci\project\tmp\$rubyArchitecture\stage\lib\tiny_tds"
+            $destination = ".\lib\tiny_tds"
+            Get-ChildItem $source -Recurse -Exclude "*.rb" | Copy-Item -Destination {Join-Path $destination $_.FullName.Substring($source.length)}
+
+      - restore_cache:
+          name: restore mssql installation file
+          key: downloads-{{ checksum "test/bin/install-mssql.ps1" }}
```

Comment:
Can you clarify the benefit of caching this installation script? It's just another source file in the repo, so it should already be available on disk in the job step.

Reply:
It doesn't cache the PowerShell script, but the `C:/Downloads` folder where I download the MSSQL installation file into. As the URL is hardcoded into this script, I thought it was fine to use the script for the checksum.

Potential side-effect is that if you change something in the script, it'll re-download the installation file. But it was also continue to use the cache even if Microsoft updates the SQL installation file on their side. I wouldn't say it's too important that the CI uses the latest version and rather take the 2 minute speed-up from caching the file.

Reply:
> It doesn't cache the PowerShell script, but the C:/Downloads folder where I download the MSSQL installation file into

Oh yeah... duh.